Subject: Fix x32 compilation issue (Closes: #970662)

Debian.org x32 builders were failing to build 1:10.5.5-1 with error message:
  operand size mismatch for `crc32'

This patch fixes that issue, as proven in
https://people.debian.org/~jrtc27/mariadb-10.5_10.5.5-1+jrtc27.1_x32-2020-09-28T18:19:31Z.build

However, the build does not yet fully pass as two test suite failures remain.

Credits: Jessica Clarke

--- a/storage/innobase/ut/ut0crc32.cc
+++ b/storage/innobase/ut/ut0crc32.cc
@@ -164,7 +164,7 @@ static inline ulint ut_crc32c_64(ulint c
 #  elif __has_feature(memory_sanitizer)
   return __builtin_ia32_crc32di(crc, data);
 #  elif defined __x86_64__
-  asm("crc32q %1, %0" : "+r" (crc) : "rm" (data));
+  asm("crc32q %1, %q0" : "+r" (crc) : "rm" (data));
   return crc;
 #  else
   asm("crc32l %1, %0" : "+r" (crc) : "rm" (static_cast<uint32_t>(data)));
